presto-main/src/main/java/io/prestosql/execution/TaskStatus.java

```diff
@@ -89,7 +94,8 @@ public TaskStatus(
             @JsonProperty("systemMemoryReservation") DataSize systemMemoryReservation,
             @JsonProperty("revocableMemoryReservation") DataSize revocableMemoryReservation,
             @JsonProperty("fullGcCount") long fullGcCount,
-            @JsonProperty("fullGcTime") Duration fullGcTime)
+            @JsonProperty("fullGcTime") Duration fullGcTime,
+            @JsonProperty("dynamicFilterDomains") Map<String, Domain> dynamicFilterDomains)
```

Comment:
@raunaqmorarka This might potentially increase QueryInfo significantly. Could you do some experiment (on some TPCH/TPCDS query with large number of joins) to see how big `QueryInfo` is with `dynamicFilterDomains` being part of `TaskStatus` compared to `TaskStatus` without `dynamicFilterDomains`. Maybe we need to simplify domains before putting them in `TaskStatus`.

Reply:
In q64 size of QueryInfo increased from 3.8MB to 6 MB running on my laptop, in actual cluster this could get much bigger with many tasks.
Pushed changes to simplify domains before putting them in TaskStatus.

Reply:
At some point, we'll have to either introduce a new mechanism to propagate task state similar to how data is propagated via exchanges (i.e., instead of propagating a cumulative state, we propagate events as they happen -- the establishment of a dynamic filter being an example of one such event). That way, we won't have to keep re-sending the filter on every task update.